### ISSUE TITLE
chmod flags not compatible with mac?

### ISSUE THREAD

**Issue by klokie** [Akianonymus/gdrive-downloader#50]
Hi, I'm trying to get this working on a Mac M1 running Mac OS Ventura (13.3.1). I've got Homebrew and used it to install both this script and bash 5.2.15. 

The script exits after the "Enter Client ID" step, with the error "chmod: --: No such file or directory". I wasn't familiar with the double dash `--` syntax, maybe it's not supported by the `chmod` provided by MacOS?

**Comment by jaysoffian** [comment]
Yup, same for FreeBSD. The script does this in case the filename starts with a `-` to prevent it being interpreted as an option. As portable way to do this would be to prefix filenames that start with a `-` with `./`.

**Comment by klokie** [comment]
> A portable way to do this would be to prefix filenames that start with a - with ./.

I tried that, but the script still exited. The only way I could get it to run was by removing all calls to `chmod`; then it worked ok.

**Comment by Akianonymus** [comment]
> 

Hmm, i will try to fix.

**Comment by jaysoffian** [comment]
> > A portable way to do this would be to prefix filenames that start with a - with ./.
> 
> I tried that, but the script still exited. The only way I could get it to run was by removing all calls to `chmod`; then it worked ok.

I didn't mean for you to prefix filenames. I meant that the script would need to be updated to remove `--` and use `./$filename` instead in cases where `$filename` does not start with `/`. The reason the `--` was put into the script was in case a filename starts with a `-` to prevent `chmod` from interpreting it as an option.

**Comment by Akianonymus** [comment]
Fixed in 1f64947 — removed the unsupported `--` delimiter from all `chmod` calls so this works on macOS and FreeBSD.
